### PR TITLE
Convert app to use one spinner attached to JFScene + resize spinner

### PR DESCRIPTION
**Pull request by cewert**
## Changes
- Use booleans on main JFScene to control loading spinner
  - `isLoading` and `disableRemote`
- Refactor from 3 to 2 main functions to control spinner
  - `startLoadingSpinner(disableRemote = true as boolean)`
    - Disables the remote as needed with a progressdialog and shows the loading spinner
    - Only display spinner when `isLoading` = false. This prevents double spinners.
  - `stopLoadingSpinner()`
    - Removes progressdialogs as needed and hides the loading spinner
- Modifies every main screne to call `stopLoadingSpinner()` after the data has been loaded and the screne is ready to be shown
- Removed custom Spinner components from all scenes and converted them to use the JFScene spinner
- Resize spinner image from 200px -> 125px and recenter for 1080p resolution. This makes our spinner a similar size and location as the video player spinner
- Rearrange things a bit in `CreateSeasonDetailsGroupByID()` so that the loading spinner goes away when the episodes are loaded instead of waiting on the extras/specials logic to finish

### CODE REVIEW

**Comment by sevenrats** [comment]
just noting here what we discussed in chat, this does not mitigate the black screen that results from slow remote backends.

**Comment by cewert** [comment]
I pretty much redid this whole PR and updated the OP. Added the main logic to JFScene, using 2 booleans on JFScene instead of the 4 component functions, and I added a `stopLoadingSpinner()` to every main screne when it was done loading data. I also converted all the current custom spinners to use the new JFScene one instead. Our whole app uses one spinner now that can be toggled by using `startLoadingSpinner()` and `stopLoadingSpinner()`.

File: components/JFScene.bs [hunk 1]
```diff
@@ -3,6 +3,31 @@ import "pkg:/source/utils/misc.bs"
 sub init()
     m.top.backgroundColor = "#262626" '"#101010"
     m.top.backgroundURI = ""
+    m.spinner = m.top.findNode("spinner")
+end sub
+
+' Triggered when the isLoading boolean component field is changed
+sub isLoadingChanged()
+    if m.top.isLoading
```
Comment (1hitsong) [inline components/JFScene.bs]:
```suggestion
    m.spinner.visible = m.top.isLoading
```

**Review by 1hitsong — CHANGES_REQUESTED** [review]
1 small code style change.

**Comment by 1hitsong** [comment]
There's still an old spinner ref on line 553 of `components\ItemGrid\MusicLibraryView.bs`

**Comment by 1hitsong** [comment]
Inside `Main.bs` there are still 5 references to m.spinner. The ones without isValid checks are throwing errors, then ones with isValid checks simply aren't showing spinners.

**Comment by cewert** [comment]
## Changes
- Addressed reviewer feedback
- Fixed all remaining custom spinner references/bugs
- Added the spinner to a few more things triggered from the main loop (selecting a series, season, album, play all artist button, instant mix button etc.)
- Stop showing the spinner a little earlier on audio player view.

**Comment by 1hitsong** [comment]
I have 1 photo in my photo library. Clicking on it results in an infinite spinner that never goes away. Additionally, I'm unable to press back to get out of the spinner. I'm stuck unless I press home.

Quick play on the photo library and on the photo works. This occurs when pressing OK from the grid.

**Comment by 1hitsong** [comment]
I pressed Play on the Recordings Library. Nothing loaded and I'm again stuck on an infinite spinner.

**Comment by sevenrats** [comment]
this now does fix the missing spinner, but that spinner is behind some darkening left over from the progressDialog for some reason.  This is not the end of the world but it would be better to fix it somehow.
cewert when you have some time, please let me know how you feel about a solution like this one:
https://github.com/cewert/jellyfin-roku/pull/6/files

**Comment by sevenrats** [comment]
also just noticed that there is still no spinner between episodes

**Comment by 1hitsong** [comment]
> I pressed Play on the Recordings Library. Nothing loaded and I'm again stuck on an infinite spinner.

This issue is still occurring. All my other issues are resolved.

Think I've got actionable info. Recordings library will quickplay in this PR if I have an unwatched video.

If I mark the one video I have as watched, then I will get stuck at an infinite spinner. I'm guessing the path when there's no video to play doesn't have a stop spinner call.

**Comment by cewert** [comment]
## Changes
- Running `stopLoadingSpinner()` whenever quickplay api quieries are invalid or there are 0 items returned.
- Don't assume item type - Copy recording library logic fix from refactored new function in  #1531

File: source/utils/quickplay.bs [hunk 2]
```diff
@@ -545,7 +571,15 @@ namespace quickplay
             print "tvshowsData=", tvshowsData
 
             if isValid(tvshowsData) and isValidAndNotEmpty(tvshowsData.items)
-                quickplay.multipleSeries(tvshowsData.items)
+                ' the type of media returned from api may change.
+                if tvshowsData.items[0].Type = "Series"
```
Comment (1hitsong) [inline source/utils/quickplay.bs]:
```suggestion
                if LCase(tvshowsData.items[0].Type) = "series"
```

**Review by 1hitsong — CHANGES_REQUESTED** [review]

**Review by 1hitsong — APPROVED** [review]

**Pull request merged**